### PR TITLE
oauth2: enable `use_refresh_token` by default

### LINKED ISSUE #36045: Question: Why is `use_refresh_token` is disabled by default in oauth2 filter?
*Title*: *Why is `use_refresh_token` [disabled by default in oauth2 filter](https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/filters/http/oauth2/v3/oauth.proto#envoy-v3-api-field-extensions-filters-http-oauth2-v3-oauth2config-use-refresh-token)?*

*Description*:
> Most of the oauth/ oidc provider have an option to select if refresh token should be sent in the response. With the `use_refresh_token` option, we have a way to disable using refresh token, however if the same is not sent in response there is no need to generally disable it. We could keep the default behavior of this flag to be `true` by default.

I wished to understand, apart from unknowingly using the refresh token and thus a security implication, what are the reasons to not have it enabled by default?

### PR DESCRIPTION
**Pull request by kanurag94**
Commit Message: enable `use_refresh_token` by default in oauth2 filter.
Additional Description: Almost all oauth provider give you an option to enable or disable sending refresh_token in response. If someone is sending refresh_token, the intent is generally to use it. If not, the same is not used and the behavior stays the same. This feature is now pretty stable and can be enabled by default.
Risk Level: Low
Testing: UTs
Docs Changes: Present in the PR.
Release Notes: Need to mention about enabling `use_refresh_token` by default.

Fixes #36045

### CODE REVIEW

**Comment by kanurag94** [comment]
@wbpcode may you please help me reopen this PR

**Comment by phlax** [comment]
@kanurag94 there is a cert expiry - see #36755

**Comment by kanurag94** [comment]
thanks @phlax for keeping me updated, will watch for it to get merged

**Comment by phlax** [comment]
merged now - apologies for disruption to ci

File: api/envoy/extensions/filters/http/oauth2/v3/oauth.proto [hunk 1]
```diff
@@ -148,7 +148,7 @@ message OAuth2Config {
 
   // If set to true, allows automatic access token refresh using the associated refresh token (see
   // `RFC 6749 section 6 <https://datatracker.ietf.org/doc/html/rfc6749#section-6>`_), provided that the OAuth server supports that.
-  // Default value is false.
+  // Default value is true.
```
Comment (wbpcode) [inline api/envoy/extensions/filters/http/oauth2/v3/oauth.proto:151]: It's ok to change the default value to a more appropriate value. But please guard this by a runtime flag.

Comment (kanurag94) [reply api/envoy/extensions/filters/http/oauth2/v3/oauth.proto:151]: Ack, I didn't think this is a high impact change. Added now

**Review by wbpcode — COMMENTED** [review]
LGTM. Thanks. Please resolve the CI and conflict.

File: api/envoy/extensions/filters/http/oauth2/v3/oauth.proto (hunk 1 above)
Comment (wbpcode) [reply api/envoy/extensions/filters/http/oauth2/v3/oauth.proto:151]: Yeah, it's not big impact, although it's should not a very minor change. So, be careful would be better anyway.

Comment (kanurag94) [reply api/envoy/extensions/filters/http/oauth2/v3/oauth.proto:151]: Agreed! Thanks a lot for taking a look at this :D

**Comment by kanurag94** [comment]
Thanks for reviewing this @wbpcode.
CI is now green

**Review by wbpcode — APPROVED** [review]

**Pull request merged**